multiqc/multiqc.py

```diff
@@ -499,9 +502,43 @@ def run(
         config.custom_css_files.extend(custom_css_files)
     config.kwargs = kwargs  # Plugin command line options
 
-    # Clean up analysis_dir if a string (interactive environment only)
-    if isinstance(config.analysis_dir, str):
-        config.analysis_dir = [config.analysis_dir]
+    # Discarding the CLI variables to make sure we use only config down below.
```

Comment:
I feel like we could probably avoid this with some more refactoring of the "mega do everything function" approach that is currently in `multiqc.py`.

eg. have dedicated functions to parse config, parse cli, then move around a config object cleanly instead.

Just a thought off the top of my head, can return to this another day in another PR..

Reply:
There was an old Carmack article where he argued in favour of not breaking up long functions. I think this is related: http://number-none.com/blow/blog/programming/2014/09/26/carmack-on-inlined-code.html
He suggests only refactoring code into a function if it's actually used multiple times, otherwise inline.

But we're going refactor this anyway to support interactive usage, plus to support config with Pydantic, so let's see how it looks.